esp/services/ws_workunits/ws_workunitsService.cpp

```diff
@@ -3179,9 +3186,32 @@ void getWorkunitCluster(IEspContext &context, const char *wuid, SCMStringBuffer
     }
 }
 
+__int64 getLogicalFileSize(IEspContext &context, const char *logicalName)
+{
+    StringBuffer userID;
+    context.getUserID(userID);
+    Owned<IUserDescriptor> userDesc;
+    if (!userID.isEmpty())
+    {
+        userDesc.setown(createUserDescriptor());
+        userDesc->set(userID, context.queryPassword(), context.querySignature());
+    }
+
+    Owned<IDistributedFile> df = queryDistributedFileDirectory().lookup(logicalName, userDesc, false,
```

Comment:
we should be careful not to make repeated calls to external services (Dali) for the same information..

createNewFileResultSet is already going to perform a Dali lookup for this file.

It would be better if createNewFileResultSet either performed the limit check,
or if there was a variety of createNewFileResultSet that took a IDistributedFile, then you could lookup as you are here, perform the check and pass 'df' to createNewFileResultSet.

Reply:
I guess I should add a createNewFileResultSet(IDistributedFile * df, const char * logicalFile, const char * cluster)

Reply:
queryLogicalName() is a member of IDistributedFile, so I don't think you need the const char * as well.